### DISCUSSION TITLE
Passthrough mode

### DISCUSSION THREAD

**Discussion by baltazartroisville** [abraunegg/onedrive#2503] [Q&A] [answered]
Hello,

I'm really happy having found this project, it's kind of my last hope :)

I've a special use case and found in the doc already that one requirement of it is indeed possible, namely the upload only option.

I want to use the device, on which i want to install the app, only as a relay. 

Every file that is placed inside a folder, in which onedrive is mounted, should directly be uploaded to onedrive. No local copy should be kept. 
Instead something like a symbolic link should be kept to show other software, that accesses that folder, which files are available. 

If there is a request to one of these files it should be downloaded and passed through to the requesting party and after that access, it should directly be deleted from the folder again, leaving only the "symbolic file link".

I guess my use case is rather rare but i still hope someone has an idea if that is somehow possible.

Thanks in advance!

**Comment by abraunegg** [comment, marked as answer]
@baltazartroisville 
You will need to build this sort of capability.

> Every file that is placed inside a folder, in which onedrive is mounted, should directly be uploaded to onedrive. No local copy should be kept.

Use: `--synchronize --upload-only --remove-source-files`
* Syncs your configured 'sync_dir' to Microsoft OneDrive online
* Upload only local data
* Post successful upload, delete local file

Potentially extend this by using `--enable-logging` so you have an auditable log file of events.

> Instead something like a symbolic link should be kept to show other software, that accesses that folder, which files are available.

You will have to develop this part. From your console log and/or log file, track what was uploaded and then deleted, extracting the path name.

Use the path name with: `--get-file-link ARG` which will return the online URL to access the file

Use this 'link' as your symbolic link and provide this to your clients

Track access of this link to download this file.


Now .. this being said, you are basically describing 'on-demand' access to OneDrive. Please read: 
* https://github.com/abraunegg/onedrive/issues/757
* https://github.com/jstaf/onedriver

**Reply by baltazartroisville** [reply]
thank you very much for your effort and taking the time, you helped me a lot! as far as i understood the usage of onedriver correctly after reading the github description it should do everything that i require, right? or do i have to combine the capabilities of your app with it?

to unclutter my mind i wrote a sorted summary of my needs. by the way, for more context: i want to use onedrive to store nextcloud files in it because server space is really limited and i dont have to pay for my onedrive cloud.

1. upload
- if a user puts a file in the local mounted onedrive folder the file should be uploaded to onedrive cloud and directly deleted afterwards, leaving only a symbolic file in the local folder.
- 
2. download
- if a user tries to access a file, the file should be download from onedrive cloud to the local folder, and after the request is finished the local copy in the mounted onedrive folder should get deleted again.

3. view
- it should be possible to view all files, which are saved in onedrive cloud, as some kind of symbolic file inside the local mounted onedrive folder, they should look and behave like normal files but if there is direct access they should get downloaded only on demand.

4. deletion
- if a user deletes a symbolic file from the mounted onedrive folder, the file should get deleted in onedrive cloud as well.

sorry for bothering you again, the onedriver app has sadly no discussion area or website and i thought maybe you (or someone else here?) knows it.

**Reply by abraunegg** [reply]
@baltazartroisville 
> or do i have to combine the capabilities of your app with it?

Technically, you could run the applications side by side - they are *different* applications after all - however it is *highly* ill advised and I do not in any way endorse or recommend that you do that.

> sorry for bothering you again, the onedriver app has sadly no discussion area or website and i thought maybe you (or someone else here?) knows it.

You are better off raising a new issue on the 'onedriver' GitHub site: https://github.com/jstaf/onedriver/issues

Raise all your points on what you want to do, + your thoughts from above. I cannot comment on how that application works, other than it providing the 'on-demand' functionality you seem to require. What it does with local files post upload or download I cannot comment on at all.
